x-pack/plugins/osquery/public/live_queries/form/index.tsx

```diff
@@ -228,39 +216,66 @@ const LiveQueryFormComponent: React.FC<LiveQueryFormProps> = ({
   const handleSavedQueryChange = useCallback(
     (savedQuery) => {
       if (savedQuery) {
-        updateFieldValues({
-          query: savedQuery.query,
-          savedQueryId: savedQuery.savedQueryId,
-          ecs_mapping: savedQuery.ecs_mapping
+        setValue('query', savedQuery.query);
+        setValue('savedQueryId', savedQuery.savedQueryId);
+        setValue(
+          'ecs_mapping',
+          !isEmpty(savedQuery.ecs_mapping)
             ? map(savedQuery.ecs_mapping, (value, key) => ({
                 key,
                 result: {
                   type: Object.keys(value)[0],
-                  value: Object.values(value)[0],
+                  value: Object.values(value)[0] as string,
                 },
               }))
-            : [],
-        });
+            : [defaultEcsFormData]
+        );
 
         if (!isEmpty(savedQuery.ecs_mapping)) {
           setAdvancedContentState('open');
         }
       } else {
-        setFieldValue('savedQueryId', null);
+        setValue('savedQueryId', null);
       }
     },
-    [setFieldValue, updateFieldValues]
+    [setValue]
   );
 
+  const onSubmit = useCallback(
+    // not sure why, but submitOnCmdEnter doesn't have proper form values so I am passing them in manually
+    async (values: ILiveQueryFormFields = watchedValues) => {
+      const serializedData = pickBy(
+        {
+          agentSelection: values.agentSelection,
+          saved_query_id: values.savedQueryId,
+          query: values.query,
+          pack_id: packId?.length ? packId[0] : undefined,
+          ...(values.ecs_mapping
+            ? { ecs_mapping: convertECSMappingToObject(values.ecs_mapping) }
+            : {}),
+        },
+        (value) => !isEmpty(value)
+      );
+      if (isEmpty(errors)) {
+        try {
+          // @ts-expect-error update types
+          await mutateAsync(serializedData);
+          // eslint-disable-next-line no-empty
+        } catch (e) {}
+      }
+    },
+    [errors, mutateAsync, packId, watchedValues]
+  );
   const commands = useMemo(
     () => [
       {
         name: 'submitOnCmdEnter',
         bindKey: { win: 'ctrl+enter', mac: 'cmd+enter' },
-        exec: () => submit(),
+        // @ts-expect-error update types - explanation in onSubmit()
+        exec: () => handleSubmit(onSubmit)(watchedValues),
```

Comment:
is there a way to avoid passing `handleSubmit(onSubmit)` in all the forms? I think it should be either coming from the `useForm()`,  or be built into `onSubmit`

Reply:
not sure if that would be possible. I can of course implement handleSubmit inside of onSubmit but I am not sure if that brings any value.